### DISCUSSION TITLE
Returning null Values in GraphQL Responses

### DISCUSSION THREAD

**Discussion by giordano-newday** [Netflix/dgs-framework#1717] [General]
I'm part of a team using DGS for our backend,  Apollo Client on Android and a custom GraphQL client on iOS. We've encountered an issue: DGS omits fields in the response when they are nullable and have a null value. This hasn't been a problem for our current setup, but we're exploring using Apollo Client for iOS, which expects these fields to be present and returns a decode error when they're absent.

From our understanding of the GraphQL spec, nullable fields should be included in responses as null. Is there a configuration in DGS that allows for this behaviour?
A link to the Apollo Client Swift question: https://github.com/apollographql/apollo-ios/issues/3216

**Comment by JanPretzel** [comment]
We are currently running into the same problem. Apollo (React client in our case) expects nullable fields that resolve to `null` should be included in the response with a `null` value rather than omitted as defined in the [GraphQL spec](https://spec.graphql.org/September2025/#sec-Value-Completion). Did you ever find a solution for this? Would using a [Custom Object Mapper](https://netflix.github.io/dgs/advanced/custom-object-mapper/) be the correct approach here?

**Reply by pascalwill** [reply]
We had a similar issue because we did define an `ObjectMapper` as `@Primary` bean which has some "optimizations" enabled. You might want to check if you already have an `ObjectMapper` defined somewhere.

**Reply by JanPretzel** [reply]
@pascalwill thanks, this seems to be our problem as well!
